appinventor/blocklyeditor/src/blocklyeditor.js

```diff
@@ -40,6 +40,31 @@ Blockly.configForTypeBlock = {
 Blockly.BlocklyEditor.render = function() {
 };
 
+unboundVariableHandler = function(myBlock, yailText) {
+  var unbound_vars = Blockly.LexicalVariable.freeVariables(myBlock);
+  unbound_vars = unbound_vars.toList();
+  if (unbound_vars.length == 0) {
+    Blockly.ReplMgr.putYail(yailText, myBlock);
+  } else {
+    var form = "<form>Enter values for:<br>";
+    for (v in unbound_vars) {
+      form  +=  unbound_vars[v] + '<input type=text name=' + unbound_vars[v] + '><br>';
+    }
+    form += "</form>";
+    var dialog = new Blockly.Util.Dialog('Unbound Variables',form, 'Submit', false, 'Cancel', 10, function (button) {
```

Comment:
You should add entries to the messages file so that these strings can be internationalized (cancel already should be).